src/chapter_1.md

```diff
@@ -1 +1,499 @@
+            random_byte: (self.metrics.as_ref().expect("Can't read byte_metric")).a_metric,
+        });
+
+        alumet.add_source(initial_source, TriggerSpec::at_interval(self.config.poll_interval));
+        Ok(())
+    }
+
+    // The stop function is called after all the metrics, sources and output previously 
+    // registered have been stopped and unregistered.
+    fn stop(&mut self) -> anyhow::Result<()> {
+        Ok(())
+    }
+}
+```
+
+As you can see, currently the start function is empty, let's fill it now.
+We want to create a new metric to match with the Metrics structure's field. In this structure, we have one field: *a_metric*.
+First, we create a unit associated with the metric:
+
+```rust
+# extern crate alumet;
+# use alumet::units::{PrefixedUnit, Unit, UnitPrefix};
+let my_byte_unit: PrefixedUnit = PrefixedUnit { base_unit: Unit::Byte, prefix: UnitPrefix::Plain };
+```
+
+Then, we use the create_metric() function of the alumet::plugin::AlumetStart. We specify the kind of value (u64), the name
+of the metric, its unit (created above) and the last argument is the description:
+
+```rust,ignore
+let byte_metric = alumet.create_metric::<u64>(
+        "random_byte",
+        my_byte_unit,
+        "Byte randomly get",
+    )?;
+self.metrics = Some(Metrics {
+    a_metric: byte_metric
+});
+```
+
+Now that we have our metric, we need to add a Source to Alumet.
+
+The ThePluginSource structure will be used as a buffer to retrieve values. We need to add this as ALUMet source:
+
+```rust,ignore
+// We create a source from ThePluginSource structure.
+let initial_source = Box::new(ThePluginSource {
+    random_byte: (self.metrics.expect("Can't read byte_metric")).a_metric,
+});
+// Then we add it to the alumet sources, adding the poll_interval value previously defined in the config.
+alumet.add_source(initial_source, TriggerSpec::at_interval(self.config.poll_interval));
+```
+
+Currently, you should have an error about your initial source, it's because the trait bound
+`ThePluginSource: alumet::pipeline::Source` is not satisfied. We are now going to implement `Source` to fix this.
+
+### Implement Source
+
+In this part, we will implement the Source trait for our ThePluginSource structure.
+
+```rust
+# extern crate serde;
+# extern crate humantime_serde;
+# extern crate anyhow;
+# extern crate alumet;
+# use alumet::{
+#     measurement::{MeasurementAccumulator, MeasurementPoint, Timestamp},
+#     metrics::TypedMetricId,
+#     pipeline::{elements::error::PollError, trigger::TriggerSpec, Source},
+#     plugin::{
+#         rust::{deserialize_config, serialize_config, AlumetPlugin},
+#         ConfigTable,
+#     },
+#     resources::{Resource, ResourceConsumer},
+#     units::{PrefixedUnit, Unit, UnitPrefix},
+# };
+# use serde::{Deserialize, Serialize};
+# use std::{fs::File, io::Read, time::Duration};
+# 
+#[derive(Debug)]
+# struct ThePluginSource {
+#     random_byte: TypedMetricId<u64>,
+# }
```

Comment:
https://github.com/alumet-dev/developer-book/pull/3/files#r1696843603

Reply:
this is not a part of file. This is the code